lib/OpenQA/Worker/CommandHandler.pm

```diff
@@ -67,18 +66,12 @@ sub handle_command {
             return undef;
         }
     }
-    elsif ($type ne 'info') {
+    elsif ($type ne 'info' && $type ne 'grab_job' && $type ne 'grab_jobs') {
```

Comment:
Maybe the condition should be the types we want?

Reply:
I had that thought as well but didn't want to re-decide about that design for the sake of this change. The current design makes basically the assumption that all message types are job-specific by default and other messages are the exception. I had to make grab job messages the exception as well because these need to be handled separately (they are job-specific but refer to a *new* job rather than the current one).

Reply:
Changed it to `exists $COMMANDS_SPECIFIC_TO_CURRENT_JOB{$type}` so it is also clear what these types have in common.